specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -0,0 +1,438 @@
+```xml
+<Extensions>
+  <com:Extension Category="windows.comServer">
+    <com:ComServer>
+      <com:ExeServer Executable="SampleBackgroundApp\SampleBackgroundApp.exe" DisplayName="SampleBackgroundApp" Arguments="-ReunionPushServer">
+        <com:Class Id="{GUIDEntryPoint}" DisplayName="Reunion Push" />
+      </com:ExeServer>
+    </com:ComServer>
+  </com:Extension>
+</Extensions>
+```
+
+## UWP InProc activator
+The UWP inproc activator flow differs from the Win32 inproc activator flow since there is technically no WinMain activation associated with UWPs. Instead the App will have to  override and implement OnBackgroundActivated which is tightly coupled to Windows.UI.Xaml workstreams.
+
+# API Details
+```c# (but really MIDL3)
+    namespace Microsoft.ProjectReunion
+    {
+        // The legacy ActivationKind does not support operations like Push. So we need to define an extension set
+        enum ActivationKindExtension
+        {
+            Launch = 0,
+            Search = 1,
+            ShareTarget = 2,
+            File = 3,
+            Protocol = 4,
+            FileOpenPicker = 5,
+            FileSavePicker = 6,
+            CachedFileUpdater = 7,
+            ContactPicker = 8,
+            Device = 9,
+            PrintTaskSettings = 10,
+            CameraSettings = 11,
+            RestrictedLaunch = 12,
+            AppointmentsProvider = 13,
+            Contact = 14,
+            LockScreenCall = 15,
+            VoiceCommand = 16,
+            LockScreen = 17,
+            PickerReturned = 1000,
+            WalletAction = 1001,
+            PickFileContinuation = 1002,
+            PickSaveFileContinuation = 1003,
+            PickFolderContinuation = 1004,
+            WebAuthenticationBrokerContinuation = 1005,
+            WebAccountProvider = 1006,
+            ComponentUI = 1007,
+            ProtocolForResults = 1009,
+            ToastNotification = 1010,
+            Print3DWorkflow = 1011,
+            DialReceiver = 1012,
+            DevicePairing = 1013,
+            UserDataAccountsProvider = 1014,
+            FilePickerExperience = 1015,
+            LockScreenComponent = 1016,
+            ContactPanel = 1017,
+            PrintWorkflowForegroundTask = 1018,
+            GameUIProvider = 1019,
+            StartupTask = 1020,
+            CommandLineLaunch = 1021,
+            BarcodeScannerProvider = 1022,
+            Push = 1023,
+        };
+
+        // Simply a wrapper around IActivatedEventArgs to support new Kind extensions like Push which aren't a part of Foundations today
+        runtimeclass ActivatedEventArgsExtension
+        {
+            ActivatedEventArgsExtension(Windows.ApplicationModel.Activation.IActivatedEventArgs activatedEventArgs, ActivationKindExtension kind2);
+            ActivationKindExtension Kind2 { get; };
```

Comment:
```suggestion
            ActivatedEventArgsExtension(Windows.ApplicationModel.Activation.IActivatedEventArgs activatedEventArgs, ActivationKind2 kind2);
            ActivationKind2 Kind2 { get; };
```
Perhaps use `Kind2` suffix, to align with the other extensions? Makes it clear what the dev needs to create/pass in and leaves wiggle room for a `Kind3` someday.

Reply:
Good point. I'm talking with the folks on the AppLifecycle side who implemented this originally to investigate the best way to fix this. They may have an alternative approach. Please stay tuned.

Reply:
@aeloros FYI

Reply:
I would prefer to avoid this naming pattern completely.

Reply:
@hulumane I finally figured out the splinter in my mind. Where is ActivationKind2 defined?

I'm not seeing ActivationKind2 defined anywhere in this spec nor currently defined in the repo. An oversight in your spec?

What namespace is this expected in? If a ProjectReunion namespace then why not `Microsoft.ApplicationModel.Activation.Kind`?

What's the relationship of Kind2 to Windows.ApplicationModel.Activation.Kind?

What's the evolutionary model for this going forward? Is Reunion's ActivationKind expected to only support values not in W.AM.A.Kind? A superset of W.AM.A + new Reunion defined values? How are both extended going forward?

Reply:
@DrusTheAxe ActivationKind2 was a suggested edit by me. Think you're confusing the _suggested_ box for the actual diff just above it.

Reply:
@DrusTheAxe  Microsoft.Application.Activation.Kind does not support Push kind types today. So in this spec, I simply added an extension to the ActivationKind to support this. However, this really needs to be part of a new spec because the problem of adding new Kinds is bound to come up again in Reunion. We have a syncup today with @aeloros and @jonwis to figure out a solution here.

Reply:
We got some consensus yesterday on how to address the GetActivationArgs issue for newer Kinds like Push. It will be part of a seperate spec. In the meantime, I will remove any inferences to idl changes related to GetActivationArgs and instead infer to the newer spec.